docs/meshstack.aws.reserved-instance-guide.md

```diff
@@ -0,0 +1,27 @@
+---
+id: meshstack.aws.reserved-instance-guide
+title: Reserved Instances & Savings Plans Guide
+---
+
+An important piece of information regarding Reserved Instances to know before following this document, quoting from the AWS documentation,
+
+> Reserved Instances are first applied to usage within the purchasing account, followed by qualifying usage in any other account in the organization.
+
+and regarding Savings Plans
+
+> In a Consolidated Billing Family, Savings Plans are applied first to the owner account's usage, and then to other accounts' usage.
+
+Because of this, there are two main approaches when purchasing Reserved Instances and Savings Plans
+
+* The Cloud Foundation team buys the Reserved Instances and Savings Plans under their own AWS Account.
+
+    Because of how AWS applies the benefit of Reserved Instances, if the cloud foundation team is not running any workload, all other AWS accounts in the organisation will have the opportunity to benefit from the purchased reserved instances and savings plans. When using this approach, charging the amortized cost of the usage is enough for the chargeback process. The same applies for Savings Plans.
+
+* The teams themselves (or meshCustomers in meshcloud terminology) pay the Cloud Foundation team upfront, and the Cloud Foundation team buys Reserved Instances and Savings Plans under the meshCustomers' tenants.
+
+    There might be meshCustomers/meshProjects that can estimate their own workload and would prefer to have Reserved Instances or Savings Plans that give them priority. This ensures them the benefit of the RI or SP rather than having to share with other tenants and only have a chance of benefiting.
+
+    In this situation, the payment to the Cloud Foundation team happens at the time of purchase, and such tenants should not be charged for the usage of the RI and SP via the amortized cost, because they have already pre-paid.
+
+    To allow for this workflow, meshStack can be configured to add a line item in the tenant usage report for the month on which the Reserved Instance or Savings Plan starts, and the amount of this line item will be equal to the upfront payment of the RI or SP. Additionally, a discount which is equal to the amortized upfront cost will be added to the report every month until the end of the RI or SP lifetime. This discount will effectively cancel out the amortized cost for the usage or non-usage of the RI or SP or provide a discount in the case of some other account utilizing it.
+
```

Comment:
The way I understood your previous explanation was

    enabling these settings allows operators to conciously deviate from the default accrual basis accounting (via amortized cost) and switch to a "cashflow basis" accounting for RI/SP usage
    this is "fair" because it moves the financial risk of low RI/SP utilization (e.g. overallocating) to the meshCustomer triggering the purchase of the RI/SP

I think it would be great if the terms accrual basis and cashflow basis appeared in the text, at least they help clarify it (to me anyway, as accrual/cashflow are established terms in accounting)

Reply:
Is our normal mode is not cashflow based? If we are talking about the cashflow between the meshCustomer and Cloud Foundation team, what we already have in place could be called cashflow based (The turp for each month shows the cashflow from meshCustomer to CloudFoundation team for that month). And for RI and SP it is the same.

And eventhough we called it `fair` chargeback initially, I wouldn't call it that now that I have more information. It doesn't really move the risk of low RI/SP utilization because the AWS Account would also be charged for[ non-usage](https://s3.us-west-2.amazonaws.com/secure.notion-static.com/7cbf8f3d-c2b0-4fcb-b3fe-e84d5908b019/Untitled.png?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=AKIAT73L2G45O3KS52Y5%2F20210831%2Fus-west-2%2Fs3%2Faws4_request&X-Amz-Date=20210831T100500Z&X-Amz-Expires=86400&X-Amz-Signature=89459059856153e46edcb9a70bf3a97498bfc4b45fa47bfae054bd8e3b883d92&X-Amz-SignedHeaders=host&response-content-disposition=filename%20%3D%22Untitled.png%22) via the amortized cost.

Reply:
To the extent that meshSTack uses amortisedCost, the "default" accounting is accrual basis as well  because amortizedCost is accrual basis cf. https://docs.aws.amazon.com/cur/latest/userguide/amortized-reservation.html

Does meshStack also import non-usage data?  For the legacy CUR import there were dedicated rules about the line items and which of these got filtered https://docs.meshcloud.io/docs/meshstack.aws.metering.html#aws-cost--usage-report-based-approach-deprecated

Reply:
Also adding to this, it's my understanding that when the cloud foundation team purchases RIs/SPs centrally, that accrual basis accounting is the only method to ensure that, after full consumption of the RI/SP, the total amount that the cloud foundation team collected via meshStack chargeback is equal to the total cost of the RI/SP. On cashflow basis, the cloud foundation team would never recoup the upfront fee. 

This is also a good point. How does reservedInstanceFairChargeback play out in a "hybrid" scenario where a cloud foundation team uses both approaches (buying RI/SPs centrally on the billing account, buying RI/SPs for individual accounts) simultaneously?

Reply:
In the link you sent `If the line item type is RIFee, we take reservation/UnusedAmortizedUpfrontFeeForBillingPeriod` . So I would say that we were actually counting non-usage costs. But the old mechanism is somewhat of a mess. It was very difficult to make sense of all the columns of the huge csv file.

I will do the change about accrual based accounting.

Reply:
I did some changes. Didn't add tables (which we discussed offline) though, because it would be difficult to communicate clearly enough with those. But I did add exmplanations on the different things the discount achieves.